nomad/blocked_evals.go

```diff
@@ -120,8 +120,9 @@ func (b *BlockedEvals) SetEnabled(enabled bool) {
 }
 
 // Block tracks the passed evaluation and enqueues it into the eval broker when
-// a suitable node calls unblock.
-func (b *BlockedEvals) Block(eval *structs.Evaluation) {
+// a suitable node calls unblock. If the block is occuring by an outstanding
+// evaluation, the evaluation's token should be passed.
+func (b *BlockedEvals) Block(eval *structs.Evaluation, token string) {
```

Comment:
It does seem like two different call paths to have vs not have the token. It seems cleaner to have a `Block` and `Reblock` path instead of an optional empty string